pylabrobot/resources/carrier.py

```diff
@@ -27,14 +29,22 @@ def assign_child_resource(
     reassign: bool = True
   ):
     self.resource = resource
-    return super().assign_child_resource(resource, location=location)
+    return super().assign_child_resource(resource, location, reassign)
 
   def unassign_child_resource(self, resource):
     self.resource = None
     return super().unassign_child_resource(resource)
+  
+  def serialize(self) -> dict:
+      return { **super().serialize(), "name": self.name}
+  
+  def __hash__(self):
+        return hash((self.name))
 
```

Comment:
is this needed over `Resource.__hash__`?

```python
  def __hash__(self) -> int:
    return hash(repr(self))
```

Reply:
will revert this, holdover from debugging.
apologies for the messy PR - realized I was stuck & you found exactly why, thank you!